encoders.go

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Julien Vehent jvehent@mozilla.com [:ulfr]
+
+package main
+
+import (
+	"encoding/asn1"
+	"encoding/base64"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+func encode(sig *ecdsaSignature, format string) (str string, err error) {
+	if len(format) < 4 {
```

Comment:
Perhaps you want to use "rs_base64url" by default? The magic number 4 is a bit confusing here.